### ISSUE TITLE
Document Info Dict data are completely ignored in `getMetadata()`

### ISSUE THREAD

**Issue by Voileexperiments** [mozilla/pdf.js#10344]
I'm trying [this pdf](https://arxiv.org/pdf/1703.01161.pdf) via #8180. There is an Document Info Dict entry in this document:

```
PTEX.Fullbanner: "This is pdfTeX, Version 3.14159265-2.6-1.40.17 (TeX Live 2016) kpathsea version 6.2.2"
```

However it is not shown inside the result of `getMetadata()`.

The problem lies within `get documentInfo()` in default `pdf.worker.js` (I don't know where these lines belong in the source, so I'm only referring to the dist version from the [online demo](https://mozilla.github.io/pdf.js/web/viewer.html)):

```javascript
get documentInfo() {
      var docInfo = {
        PDFFormatVersion: this.pdfFormatVersion,
        IsLinearized: !!this.linearization,
        IsAcroFormPresent: !!this.acroForm,
        IsXFAPresent: !!this.xfa
      };
      var infoDict;

      try {
        infoDict = this.xref.trailer.get('Info');
      } catch (err) {
        if (err instanceof _util.MissingDataException) {
          throw err;
        }

        (0, _util.info)('The document information dictionary is invalid.');
      }

      if ((0, _primitives.isDict)(infoDict)) {
        for (var key in DocumentInfoValidators) {
          if (infoDict.has(key)) {
            var value = infoDict.get(key);

            if (DocumentInfoValidators[key](value)) {
              docInfo[key] = typeof value !== 'string' ? value : (0, _util.stringToPDFString)(value);
            } else {
              (0, _util.info)('Bad value in document info for "' + key + '"');
            }
          }
        }
      }

      return (0, _util.shadow)(this, 'documentInfo', docInfo);
    },
```

And earlier in the code, `DocumentInfoValidators`:

```javascript
var DocumentInfoValidators = {
    Title: _util.isString,
    Author: _util.isString,
    Subject: _util.isString,
    Keywords: _util.isString,
    Creator: _util.isString,
    Producer: _util.isString,
    CreationDate: _util.isString,
    ModDate: _util.isString,
    Trapped: _primitives.isName
  };
```

`getMetadata` calls this:

```
handler.on('GetMetadata', function wphSetupGetMetadata(data) {
      return Promise.all([pdfManager.ensureDoc('documentInfo'), pdfManager.ensureCatalog('metadata')]);
    });
```

However, the `metadata` here refers to the catalog dict, not the info dict:

```
{
    key: "metadata",
    get: function get() {
      var streamRef = this.catDict.getRaw('Metadata');

      if (!(0, _primitives.isRef)(streamRef)) {
        return (0, _util.shadow)(this, 'metadata', null);
      }

      var suppressEncryption = !(this.xref.encrypt && this.xref.encrypt.encryptMetadata);
      var stream = this.xref.fetch(streamRef, suppressEncryption);
      var metadata;

      if (stream && (0, _primitives.isDict)(stream.dict)) {
        var type = stream.dict.get('Type');
        var subtype = stream.dict.get('Subtype');

        if ((0, _primitives.isName)(type, 'Metadata') && (0, _primitives.isName)(subtype, 'XML')) {
          try {
            metadata = (0, _util.stringToUTF8String)((0, _util.bytesToString)(stream.getBytes()));
          } catch (e) {
            if (e instanceof _util.MissingDataException) {
              throw e;
            }

            (0, _util.info)('Skipping invalid metadata.');
          }
        }
      }

      return (0, _util.shadow)(this, 'metadata', metadata);
    }
  }
```

Basically, from how it's written currently, only the 4 fields plus all the *required* fields in document info are outputted to the `docInfo` dictionary. Any other metadata in `infoDict` is dropped.

---

I'm not sure if this is intended or not, but a fix would be changing `var key in DocumentInfoValidators` to `var key in Object.assign({}, DocumentInfoValidators, infoDict)`. Again, I don't know if this is intended behaviour, and even if it's not, I don't know where this belongs in the source code either so I can't make a PR.

**Comment by Voileexperiments** [comment]
Made a quick PR: https://github.com/mozilla/pdf.js/pull/10345

**Comment by Snuffleupagus** [comment]
Duplicate of #5970.

**Comment by Voileexperiments** [comment]
It's not a dupe, I've already seen that issue before I posted this: that uploader uses v0.x of pdf.js where everything is different and even `pdf.worker.js` doesn't exist.
